### PR TITLE
[SYCL] Fix asynchronous exceptions handling

### PR DESCRIPTION
**Pull request by Fznamznon**
This fixes following problem:
If provided async_handler throws an exceptions from passed exception
list, this list won't be cleared and exceptions will be thrown again
from queue destructor.

Signed-off-by: Mariya Podchishchaeva <mariya.podchishchaeva@intel.com>

### CODE REVIEW

File: sycl/include/CL/sycl/detail/queue_impl.hpp [hunk 1]
```diff
@@ -122,9 +122,10 @@ class queue_impl {
 
   void throw_asynchronous() {
     if (m_AsyncHandler && m_Exceptions.size()) {
-      m_AsyncHandler(m_Exceptions);
+      exception_list Exceptions;
+      std::swap(m_Exceptions, Exceptions);
+      m_AsyncHandler(Exceptions);
```
Comment (romanovvlad) [inline sycl/include/CL/sycl/detail/queue_impl.hpp]: Why not
m_AsyncHandler(std::move(Exceptions));
?

Comment (Fznamznon) [reply sycl/include/CL/sycl/detail/queue_impl.hpp]: Done, thanks!

Comment (romanovvlad) [reply sycl/include/CL/sycl/detail/queue_impl.hpp]: What about just
m_AsyncHandler(std::move(**m_**Exceptions));
?

Comment (romanovvlad) [reply sycl/include/CL/sycl/detail/queue_impl.hpp]: Disregard my latest comment.

**Pull request merged**